project/CONTRIBUTING.md

```diff
@@ -186,6 +186,25 @@ interface BarInterface
 }
 ```
 
+In some cases, you will have the possibility to warn the user that things will change,
+and recommend a new way of doing things. You can do so by triggering the dedicated kind of error, like this:
+
+```php
+<?php
+if (/* some condition showing the user is using the legacy way */) {
+    @trigger_error(
+        'Some message about this usage being deprecated in favor of something else'.
```

Comment:
* `.` -> `,`
* Maybe we can put proper examples like `'The '.__METHOD__.' method is deprecated since 3.x, to be removed in 4.0'`.
Or maybe links to good examples present on our code?